src/ia-anniversary-banner.ts

```diff
@@ -0,0 +1,194 @@
+
+  @property({ type: Number }) hideBannerDays = 7;
+
+  @property({ type: String }) directory = '';
+
+  updated(changed: any) {
+    if (changed.has('iaFiles') || changed.has('iaMD')) {
+      this.shuffledMoments = [];
+      const { directory } = this.iaMD;
+      this.directory = directory;
+    }
+    if (changed.has('directory')) {
+      this.shuffleMoments(this.directory);
+    }
+  }
+
+  shuffleMoments(directory: string) {
+    const moments: TimelineMoment[] = formatMoments(
+      this.iaFiles,
+      this.iaMD,
+      this.landingURL,
+      directory
+    );
+
+    const newMoments = [...moments];
+    this.shuffledMoments = newMoments;
+
+    this.dispatchEvent(
+      new CustomEvent('momentsShuffled', {
+        detail: { moments: this.shuffledMoments },
+      })
+    );
+  }
+
+  private closeBanner() {
+    this.viewMode = 'closed';
+
+    if (this.hideBannerDays > 0) {
+      (window as any)?.Cookies.set('anniv-banner', 'x', {
+        path: '/',
+        expires: this.hideBannerDays,
+        domain: '.archive.org',
+      });
+    }
+  }
+
+  get closeButton() {
+    return html`<button
+      class="close-banner"
+      title="close banner for the day"
+      @click=${() => this.closeBanner()}
+    >
+      ${CloseCircleIcon}
+    </button>`;
+  }
+
+  render() {
+    if (this.viewMode === 'closed') {
+      return nothing;
+    }
+
+    return html`
+      <section class=${this.viewMode}>
+        <a
+          class="left-anchor"
+          href=${this.landingURL}
+          target="_blank"
+          title="Celebrate Internet Archive's 25th Anniversary"
+          rel="nofollow"
+        >
+          ${wayforwardSvgMin} ${wayforwardSvg}
+        </a>
+        <mar-quee
+          .list=${this.shuffledMoments}
+          .defaultLink=${this.landingURL}
+        ></mar-quee>
+        ${this.closeButton}
+      </section>
+    `;
+  }
+
+  static get styles() {
+    const bannerHeight = css`var(--bannerHeight, 50px)`;
+    const marqueeWidth = css`var(--marquee-width, 150%)`;
+    const marqueeAnimation = css`var(--marquee-animation-s, 50s)`;
+
+    return css`
+      :host {
+        --bannerHeight: ${bannerHeight};
+        --marquee-height: ${bannerHeight};
+        --marquee-width: ${marqueeWidth};
+        --marquee-animation-s: ${marqueeAnimation};
+      }
+
+      section {
+        display: flex;
+        align-items: center;
+        padding: 25px;
+        color: var(--anniv-banner-text-color, #fff);
+        background-color: var(--anniv-banner-bg-color, blue);
+        padding: 0;
+        height: 0;
+      }
+      section.open {
+        height: ${bannerHeight};
+      }
+
+      section > * {
+        display: inline-block;
+        height: inherit;
+      }
+
+      .left-anchor {
+        max-width: 157px; /* width of button svg */
+        display: flex;
+        align-items: center;
+        padding: 0 10px;
+      }
+
+      mar-quee {
+        width: calc(100% - 187px);
```

Comment:
Where is the `187px` coming from? Is this related to the `157px` above? If so, you can pull out the width to its own css variable, use that for both spots, and do your calc based on that. ie:

```
const leftAnchorWidth = css`var(--leftAnchorWidth, 157px)`;
const someMargin = css`var(--leftAnchorWidth, 30px)`;
...
width: calc(100% - ${leftAnchorWidth} - ${someMargin});
```

Reply:
thanks. i've simplified so width is inherited, no longer needed